File: Android_CustomAlertDialog/AlertDialog_Homework/app/src/main/java/org/techtown/alertdialog_homework/MainActivity.java
package org.techtown.alertdialog_homework;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;


public class MainActivity extends AppCompatActivity {

    CustomAlertDialog csAlertDialog;

    String CHECK_TYPE = "Check";
    String EDIT_TYPE = "Edit";
    String ALERT_TYPE = "Alert";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btn_checkType = findViewById(R.id.btn_checkType);
        btn_checkType.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showCustomDialog(CHECK_TYPE);
            }
        });

        Button btn_editType = findViewById(R.id.btn_editType);
        btn_editType.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showCustomDialog(EDIT_TYPE);
            }
        });

        Button btn_alertType = findViewById(R.id.btn_alertType);
        btn_alertType.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showCustomDialog(ALERT_TYPE);
            }
        });
    }

    public void showCustomDialog(String type) {
        if (type == "Check") {

            csAlertDialog = new CustomAlertDialog(this);
            csAlertDialog.setType(CHECK_TYPE);
            csAlertDialog.show();

        } else if (type == "Edit") {

            csAlertDialog = new CustomAlertDialog(this);
            csAlertDialog.setType(EDIT_TYPE);
            csAlertDialog.show();

        } else {

            csAlertDialog = new CustomAlertDialog(this);
            csAlertDialog.setType(ALERT_TYPE);
            csAlertDialog.show();

        }
    }
}

File: Mvvm_Retrofit_Room_Repository_GoogleBooksAPI/app/src/main/java/org/techtown/mvvm_retrofit_room_repository_googlebooksapi/activity/MainActivity.kt
package org.techtown.mvvm_retrofit_room_repository_googlebooksapi.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import org.techtown.mvvm_retrofit_room_repository_googlebooksapi.R
import org.techtown.mvvm_retrofit_room_repository_googlebooksapi.adapter.GoogleBooksAdapter
import org.techtown.mvvm_retrofit_room_repository_googlebooksapi.databinding.ActivityMainBinding
import org.techtown.mvvm_retrofit_room_repository_googlebooksapi.repository.Repository
import org.techtown.mvvm_retrofit_room_repository_googlebooksapi.viewmodel.MainViewModel

class MainActivity : AppCompatActivity() {

    // private val TAG = "Response"

    // lateinit var로 했을 경우 늦은초기화를 할 수 없다는 오류가 발생함(by lazy로 변경함) -> viewModel을 사용할 때 값을 넣어주지 않았음
    //private val viewModel by lazy {MainViewModel(Repository())}
    private lateinit var viewModel: MainViewModel
    private lateinit var binding: ActivityMainBinding
    //val viewModel: MainViewModel by viewModels()


    private val bookAdapter by lazy {GoogleBooksAdapter()}

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main)

        // 데이터 바인딩
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.lifecycleOwner = this

        // 어댑터 연결
        val adapter = bookAdapter
        binding.recyclerView.adapter = adapter
        binding.recyclerView.layoutManager = LinearLayoutManager(this)
        // lateinit을 해줬기 때문에 viewModel에 어떤 값이 들어갈지 넣어줘야함
        viewModel = MainViewModel(Repository())
        viewModel.makeApiCall()
        viewModel.myResponse.observe(this, Observer {
            adapter.setData(it)
        })

    }
}